catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/validate/QueryValidatorsById.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.validate;
+
+import ddf.catalog.validation.QueryValidator;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QueryValidatorsById {
```

Comment:
What is the purpose of this class?

Reply:
It is an aggregation of all `QueryValidator`s in the system that allows me to select which one I want by it's id. It's main use is to encapsulate all the karaf/bundle logic as `QueryValidator`s are added or removed from the system to simplify the logic in `CqlValidateHandler`.